sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -86,6 +102,14 @@ case class Mode(
     }.getOrElse(buffer.maxBy(_._2))._1
   }
 
+  private def isCollatedString(child: Expression): Boolean = {
+    child.dataType match {
+      case s: StringType if s.collationId != CollationFactory.UTF8_BINARY_COLLATION_ID => true
+      // maybe use supportsBinaryEquality or something else
+      case _ => false
+    }
+  }
+
```

Comment:
we generally shouldn't use collationIds to check equality, you correctly noticed that we should go with `supportsBinaryEquality` instead - you can remove this method, and perform the check above as suggested